### ISSUE TITLE
Unused @throws should not warn about descriptive annotations 

### ISSUE THREAD

**Issue by marcospassos** [pepakriz/phpstan-exception-rules#47]
I'd like to suggest introducing an option to allow descriptive `@throws` annotation for unchecked exceptions to cover several cases where is helpful to provide a description of the reason an exception is thrown.

Example:
```php
class DivisionByZeroException extends LogicException {
}

/*
 * @throws DivisionByZeroException Thrown when there is an attempt to divide an integral value by zero.
 */
function divide(int $a, int $b) {
    if ($b === 0) {
        throw new DivisionByZeroException();
    }
}
```

**Comment by pepakriz** [comment]
I have a different opinion. `@throws` annotations should be used for runtime exception only. If you want to explain input restrictions, you should use classic method description. Do you agree?

**Comment by marcospassos** [comment]
I don't think so as the `@throws` tag exists exactly for this purpose.  Not because an exception is unchecked that a method should not document that in a standardized fashion - it only makes it optional. Furthermore, it seems to be a common practice,  takes Java as an example.

Because not everyone may desire such behaviour, I suggest implementing a new option that fulfils both requirements. Frankly speaking, at Croct it's becoming impractical to continue using the library as is: we already have dozens of `ignore` entries to bypass those messages. 

I'd be glad to provide a PR if welcome. I just need to figure out how to detect if an exception has a description or not.

**Comment by pepakriz** [comment]
Resolved by https://github.com/pepakriz/phpstan-exception-rules/pull/54
